akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -550,3 +556,51 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
   // When the stage starts, start the source
   override def preStart() = startGraph()
 }
+
+object RestartWithBackoffFlow {
+
+  /**
+   * Temporary attribute that can override the time a [[RestartWithBackoffFlow]] waits
+   * for a failure before cancelling.
+   *
+   * See https://github.com/akka/akka/issues/24529
+   *
+   * Will be removed if/when cancellation can include a cause.
+   */
+  @ApiMayChange
+  case class Delay(duration: FiniteDuration) extends Attribute
+
+  /**
+   * Returns a flow that is almost identity but delays propagation of cancellation from downstream to upstream.
+   *
+   * Once the down stream is finish calls to onPush are ignored.
+   */
+  private def delayCancellation[T](duration: FiniteDuration): Flow[T, T, NotUsed] =
+    Flow.fromGraph(new DelayCancellationStage(duration))
+
+  final class DelayCancellationStage[T](delay: FiniteDuration) extends SimpleLinearGraphStage[T] {
```

Comment:
Shouldn't the attribute be used also?

Reply:
Is that possible? 

Because this is an internal stage and it goes before the user flow we're wrapping will it see the attributes on the outside RestartFlow or the wrapped Flow?

Reply:
There is a `subFusingMaterializer.runOrSomething(graph, attributes)` that you can use when materializing to provide outer/default attributes. I'd probably do that and put the default `Delay` attribute in the `initialAttributes` of `RestartWithBackoffFlow` instead, as the `getOrElse(default)` is a bit secret/hidden inside the impl, but initialAttributes will end up in debug prints etc.

Reply:
Although, that is a separate issue. So we can ignore that for now, I created #24810 to track that.

The Delay attribute moving around was more a thought than something that must be changed, I think this is fine unless you want to change it @chbatey

Reply:
As i don't really have an intuition about what is better why don't we let the other reviewer decide

Reply:
I have no clue (or rather not swap in attributes stuff)